zipline/assets/continuous_futures.pyx

```diff
@@ -403,6 +403,24 @@ cdef class OrderedContracts(object):
         else:
             return None
 
+    cpdef offset_of_contract(self, long_t sid, long_t dt_value):
+        """
+        Get the offset of the given contract relative to the contract with an
+        offset of zero on the given date.
+        """
+        current_contract_sid = self.contract_before_auto_close(dt_value)
+        if sid < current_contract_sid:
+            raise ValueError(
+                'The given sid has already expired by the given date!',
+            )
+        offset = 0
```

Comment:
Should set a type for `offset` for performance reasons.  Same with `current_contract_sid`